### PR TITLE
KProve architecture extension: made pretty printing available to...

### PR DESCRIPTION
**Pull request by denis-bogdanas**
... implementations of Rewriter. Used in later PRs.

### CODE REVIEW

**Comment by ehildenb** [comment]
@denis-bogdanas it's hard to know if an architectural change makes sense without seeing how it's used.

Would it be possible to just extend `KPrint.java` instead of making a new class? That was intended to be where all pretty and non-pretty printing functionality goes.

**Comment by denis-bogdanas** [comment]
@ehildenb If you look at PrettyPrinter, you'll see that it incapsulates 3 objects. One of them is KPrint, the other 2 judging by your design choces don't belong to KPrint. Hence the need to create a new class.

File: kernel/src/main/java/org/kframework/unparser/PrettyPrinter.java
Comment (ehildenb) [inline kernel/src/main/java/org/kframework/unparser/PrettyPrinter.java:2]: copyright on first line

File: kernel/src/main/java/org/kframework/unparser/WantsPrettyPrinter.java
Comment (ehildenb) [inline kernel/src/main/java/org/kframework/unparser/WantsPrettyPrinter.java:2]: copyright on first line

**Review by ehildenb — CHANGES_REQUESTED** [review]
some nits

File: kernel/src/main/java/org/kframework/unparser/WantsPrettyPrinter.java
Comment (denis-bogdanas) [reply kernel/src/main/java/org/kframework/unparser/WantsPrettyPrinter.java:2]: I remember there was a check for copyright at build time. Why it was not triggered?

**Review by ehildenb — APPROVED** [review]

**Comment by dwightguth** [comment]
I have reverted the merge of this PR because I needed to request changes to it but it was merged without waiting for my review. In the future please do not merge changes into this repo without my approval. If you remake the PR, I will add the comments that I wanted to make to the code.

**Pull request merged**
